### PR TITLE
fix:(PersistentUnorderedMap) JSDoc documentation fix

### PR DESCRIPTION
**Pull request by luciotato**
in the JSDoc documentation examples, c/PersistentMap/PersistentUnorderedMap
fix:(PersistentUnorderedMap) alter .delete(key) to make fewer operations
fix:(PersistentUnorderedMap) move tests to it's own test spec file - add test: 'delete last key'

### CODE REVIEW

File: assembly/__tests__/runtime/runtime.spec.ts [hunk 1]
```diff
@@ -665,156 +665,6 @@ describe("Deque should handle", () => {
   });
 });
 
-let unOrderedMap: PersistentUnorderedMap<string, TextMessage>;
```
Comment (luciotato) [inline assembly/__tests__/runtime/runtime.spec.ts:668]: moved all unordered map tests to its own test spec file

File: assembly/__tests__/runtime/unordered-map.spec.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+      true,
+      "Map does not contain a key that should be there after deletion of another key"
+    );
+    expect(unOrderedMap.getSome("mapKey1")).toStrictEqual(
+      message,
+      "Incorrect result from map get after delete"
+    );
+  });
+
+  it("deleting last entry", () => {
```
Comment (luciotato) [inline assembly/__tests__/runtime/unordered-map.spec.ts:89]: new test added

File: assembly/sdk/collections/persistentUnorderedMap.ts [hunk 3]
```diff
@@ -196,23 +196,28 @@ export class PersistentUnorderedMap<K, V> {
    * @param key Key to remove.
    */
   delete(key: K): void {
-    // make sure the item is in the set
```
Comment (luciotato) [inline assembly/sdk/collections/persistentUnorderedMap.ts:199]: faster delete when we're deleting the last (lenght-1) key
also get index && check key exists in the same operation

**Review by willemneal — APPROVED** [review]
LGTM thanks!

**Pull request merged**